### PR TITLE
rbnacl 5.x

### PR DESCRIPTION
**Pull request by nijikon**

### CODE REVIEW

**Comment by nijikon** [comment]
Ping

**Comment by mfazekas** [comment]
What is the reason for dropping old ruby version? As such change is binary incompatible we can do it with a major version change.

**Comment by nijikon** [comment]
@mfazekas rbnacl-5.0.0 requires ruby version >= 2.2.6

**Comment by mfazekas** [comment]
But net-ssh works without rbnacl and with older rbnacl, so there is probably no need to drop old rubies

File: lib/net/ssh/authentication/ed25519.rb [hunk 1]
```diff
@@ -1,4 +1,4 @@
-gem 'rbnacl', '>= 3.2.0', '< 5.0'
+gem 'rbnacl', '>= 3.2.0'
```
Comment (mfazekas) [inline lib/net/ssh/authentication/ed25519.rb]: i'm not sure if we want to allow any future rbnacl, in theory the might not be compatible, on the otherhand they'll be likely compatible for what we use.

**Comment by nijikon** [comment]
I think that we should not encourage people to use ruby versions that are not supported by the ruby-core team any longer and does not receive any fixes.

File: lib/net/ssh/authentication/ed25519.rb (hunk 1 above)
Comment (nijikon) [reply lib/net/ssh/authentication/ed25519.rb]: If it works I don't see how we should freeze it; maybe we can block it `'>= 3.2.0', '< 6.0'`. What do you think?

**Comment by mfazekas** [comment]
> I think that we should not encourage people to use ruby versions that are not supported by the ruby-core team any longer and does not receive any fixes.

It's true, but we cant drop ruby releases except in major revisions as we try to follow SemVer.

**Comment by nijikon** [comment]
I agree, so can we bump the major revision?

**Comment by nijikon** [comment]
I can extract ruby version drop to a separate PR if that helps.

**Comment by mfazekas** [comment]
> I can extract ruby version drop to a separate PR if that helps.

Yes please! I usually drop ruby versions when we release a new major version, not the other way around. I think relaxing rbnacl version req and dropping old ruby releases are separate issues

**Pull request closed without merging**